uk.co.compucorp.civicrm.hrcore/tests/phpunit/CRM/HRCore/Service/CiviHRStatsGathererTest.php

```diff
@@ -214,20 +213,56 @@ public function testContactSubtypeCountWillMatchExpectedCount() {
   }
 
   public function testDeletedEntitiesWillNotBeIncluded() {
+    $contactID = ContactFabricator::fabricate()['id'];
     $existingContactCount = $this->getCurrentContactCount();
-    $deletedContact = ContactFabricator::fabricate();
-    $deletedContactID = $deletedContact['id'];
-    civicrm_api3('Contact', 'delete', ['id' => $deletedContactID]);
-    $params = ['is_deleted' => 1, 'id' => $deletedContactID];
-    $stillExists = civicrm_api3('Contact', 'getcount', $params) == 1;
-
-    if (!$stillExists) {
-      $this->markTestSkipped('There was an error in soft deletion of contact');
-    }
+    TaskTypeFabricator::fabricate();
+    $this->setUpLeaveRequest($contactID);
+    $this->registerCurrentLoggedInContactInSession($contactID);
+    civicrm_api3(
+      'UFMatch',
+      'delete',
+      ['id' => UFMatchFabricator::fabricate()['id']]
+    );
+    civicrm_api3(
+      'Contact',
+      'delete',
+      ['id' => ContactFabricator::fabricate()['id']]
+    );
+    civicrm_api3(
+      'Task',
+      'delete',
+      ['id' => TaskFabricator::fabricate()['id']]
+    );
+    civicrm_api3(
+      'Assignment',
+      'delete',
+      ['id' => AssignmentFabricator::fabricate()->id]
+    );
+    civicrm_api3(
+      'Document',
+      'delete',
+      ['id' => DocumentFabricator::fabricate()->id]
+    );
+    civicrm_api3(
+      'LeaveRequest',
+      'delete',
+      ['id' => $this->fabricateLeaveRequest($contactID)->id]
+    );
+    civicrm_api3(
+      'HRVacancy',
+      'delete',
+      ['id' => HRVacancyFabricator::fabricate()['id']]
+    );
 
     $stats = $this->getGatherer()->gather();
 
     $this->assertEquals($existingContactCount, $stats->getEntityCount('contact'));
+    $this->assertEquals(0, $stats->getEntityCount('drupalUser'));
```

Comment:
Something I realized just now. Even though the chances of switching drupal to something else is almost 0, I don't think we should call the entity `drupalUser`. A  more generic name like `cmsUser` or simply `user` would be better

Reply:
I agree, the spec called for counting "Number of users (Drupal)", but CMS user is better and we can label it how we like when displaying it later.